packages/checkout-ui-extensions-react/src/hooks/extension.ts

```diff
@@ -0,0 +1,17 @@
+import type {Extension} from '@shopify/checkout-ui-extensions';
+
+import {useApi} from './api';
+
+/**
+ * Returns the metadata about the extension.
+ */
+export function useExtension(): Extension {
+  return useApi().extension as Extension;
+}
+
+/**
+ * Returns the metadata about the extension.
+ */
+export function useExtensionData(): Extension {
+  return useExtension();
+}
```

Comment:
Don't mark this as deprecated yet. This code deviates a bit from what we have in `ui-extensions@2023.4` to fix a doc gen issue it would produce. There will be a follow-up PR to apply this latest fix to `ui-extensions@2023.4`.